### PR TITLE
2d Color Code entry with potential figure

### PR DESCRIPTION
**Pull request by viola1963**
Added an entry for the 2D color Code with a potential figure design.
## Checklist:

I remembered to:

- [yes ] Include relevant citations I could think of (with `\cite{...}`)

- [ yes] Create links to the other referenced codes (with
      `\hyperref[code:...]{...}`)

- [ yes] Update the relevant meta changelog fields with my user_id (see
      `users/users_db.yml`; add yourself in the PR if you aren't there already)

### CODE REVIEW

File: codes/quantum/qubits/stabilizer/topological/2d_color_code.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+introduced: '\cite{arxiv:quant-ph/0605138}'
+
+description: |
+  The 2-dimensional color codes are strong \hyperref[code:css]{CSS stabilizer} codes \cite{arXiv:1407.5103} and a case of D-dimensional color codes. The 2-dimensional color code is defined on a 2-dimensional trivalent lattice with 3-colorable faces, for example a honeycomb or 4.8.8 lattice. Each face corresponds to two stabilizer generators, X and Z, and are the product of Xs acting on all the qubits of a face or Z acting on all of the qubits of a face. String operators are defined on the paths along the edges of the qubit. These paths can have branching points. Each path has two string operators, one corresponding to the X basis and one corresponding to the Z basis. Like the faces, String operators also come in 3 colors. String operators commute or anti-commute. They anti-commute when they cross an odd number of times and have a different color and type. String operators correspond to encoded X or Z operators when they are closed and not boundaries. Color codes encode twice as many qubits as the surface code.\cite{arXiv:quant-ph/0605138, doi:10.1103/PhysRevLett.98.160502, doi:10.1103/PhysRevA.76.012305}
```
Comment (valbert4) [inline codes/quantum/qubits/stabilizer/topological/2d_color_code.yml:14]: Split up into paragraphs discussing aspects of this. One giant paragraph is a bit hard to follow. See reuters.com news articles for examples of how others do this.

File: codes/quantum/qubits/stabilizer/topological/2d_color_code.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  \begin{figure}
+    \includegraphics{colorCodeHoneycombHighlightedChecks.svg}
+    \caption{Stabilizer generators and string operators of a 2D color code defined on a honeycomb lattice on a torus. 
+    The plaquette operators generate the stabilizer group of the toric code where each face corresponds to an X or Z plaquette operator.  The string operators
+    are pairs of X and Z logical operators that wrap around the torus.}
+    \label{figure:toric-code-operators}
+  \end{figure}
+protection: 'As with the surface code, the code distance depends on the specific kind of lattice used to define the code. More precisely, the distance depends on the homology of logical string operators \cite{arXiv:1311.0277}.'
```
Comment (valbert4) [inline codes/quantum/qubits/stabilizer/topological/2d_color_code.yml:22]: Can you find a specific family for some lattice of size L x L? Surface codes have such famous families, and it would be nice to compare.

Comment (viola1963) [reply codes/quantum/qubits/stabilizer/topological/2d_color_code.yml:22]: I could bring up the family of triangular codes, there is one for honeycomb and 488 lattice, I was trying to think of a way to incorporate them and whether they should be another entry

**Comment by valbert4** [comment]
Closing this in prep for new version taking care of my suggested edits
